### PR TITLE
64 feature eldritch filecompress

### LINKED ISSUE #64: [feature] Eldritch file.compress
**Is your feature request related to a problem? Please describe.**
Eldritch doesn't have a way to compress a file.

**Describe the solution you'd like**
Create a function file.compress.

Should take a file path and an archive path.
Should create a compressed archive from the source path.
Should be thoroughly tested and work on Linux, Windows, and Mac.

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?
/kind feature
/kind eldritch-function

#### What this PR does / why we need it:
Add file compression function to eldritch.
Uses the `flate2` crate to perform gzip compression.
If the source is a directory it will use the `tar` crate to create a tar ball in a temporary file first.

#### Which issue(s) this PR fixes:
Fixes #64

### CODE REVIEW

**Review by Cictrone — APPROVED** [review]
sheesh

File: implants/eldritch/Cargo.toml [hunk 1]
```diff
@@ -17,3 +17,6 @@ tokio = { version = "1.19.1" , features = ["macros"] }
 tokio-stream = "0.1.9"
 serde = { version = "1.0", features = ["derive"] }
 gazebo.version = "0.8.1"
+lzma-rs = "0.2.0"
```
[the remark this answers is not in the document]
Comment (hulto) [reply implants/eldritch/Cargo.toml:20]: Nothing! Good catch

File: implants/eldritch/src/file/compress_impl.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+use anyhow::Result;
+use flate2::{Compression};
+use tar::{Builder, HeaderMode};
+use tempfile::NamedTempFile;
+
+fn tar_dir(src: String, dst: String) -> Result<String> {
+    let src_path = Path::new(&src);
+
+    // Create the tar bulider
+    let tmp_tar_file = File::create(dst.clone()).unwrap();
```
[the remark this answers is not in the document]
Comment (hulto) [reply implants/eldritch/src/file/compress_impl.rs:12]: Handling an error now and returning anyhow error if file create fails.
Using that neat match arm pattern too. 😎

File: implants/eldritch/src/file/compress_impl.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    let mut tar_builder = Builder::new(tmp_tar_file);
+    tar_builder.mode(
+        HeaderMode::Deterministic
+    );
+
+    // Add all files from source dir with the name of the dir.
+    let _ = tar_builder.append_dir_all(
+        src_path.clone().file_name().unwrap().to_str().unwrap(),
```
[the remark this answers is not in the document]
Comment (hulto) [reply implants/eldritch/src/file/compress_impl.rs:20]: Added an error handle here.

**Pull request merged**